suite-native/module-add-accounts/src/hooks/useAddCoinAccount.ts

```diff
@@ -52,23 +68,34 @@ export const useAddCoinAccount = () => {
 
     const availableNetworkAccountTypes = useMemo(() => {
         // first account type for every network is set to normal and represents default type
-        const availableTypes: Map<NetworkSymbol | undefined, AccountType[]> = new Map();
-        Object.keys(networks).forEach(symbol =>
+        const availableTypes: Map<NetworkSymbol | undefined, [AccountType, ...AccountType[]]> =
+            new Map();
+
+        Object.keys(networks).forEach(symbol => {
+            // for Cardano only allow latest account type and coinjoin is not supported
+            const types = (Object.keys(networks[symbol].accountTypes) as AccountType[]).filter(
+                accountType => accountType !== 'coinjoin',
+            );
+
             availableTypes.set(symbol as NetworkSymbol, [
                 NORMAL_ACCOUNT_TYPE,
-                ...(Object.keys(networks[symbol].accountTypes) as AccountType[]).filter(
-                    accountType => accountType !== 'coinjoin', // coinjoin is not supported
-                ),
-            ]),
-        );
+                ...(symbol === 'ada' ? [] : types),
+            ]);
+        });
+
         return availableTypes;
     }, []);
 
-    const getAvailableAccountTypesForNetwork = ({ network }: { network: Network }) =>
-        availableNetworkAccountTypes.get(network.symbol) ?? [NORMAL_ACCOUNT_TYPE];
+    const getAvailableAccountTypesForNetwork = useCallback(
+        ({ network }: { network: Network }) =>
+            availableNetworkAccountTypes.get(network.symbol) ?? [NORMAL_ACCOUNT_TYPE],
+        [availableNetworkAccountTypes],
+    );
 
-    const getDefaultAccountType = ({ network }: { network: Network }) =>
-        getAvailableAccountTypesForNetwork({ network })[0];
+    const getDefaultAccountType = useCallback(
```

Comment:
Why are you adding back these `useCallbacks`?

Reply:
good catch, for some reason I needed addCoinAccount to be callback and it bubbled down, but now it is not needed again.

Reply:
5c5f2af2c4